test/over_react/component/lazy_test.dart

```diff
@@ -0,0 +1,663 @@
+@TestOn('browser')
+@JS()
+library rmui.test.unit.component.lazy_test;
+
+import 'dart:convert';
+import 'dart:html';
+
+import 'package:js/js.dart';
+import 'package:over_react/js_component.dart';
+import 'package:over_react/over_react.dart';
+import 'package:react/react_client.dart';
+import 'package:react/react_client/react_interop.dart' hide lazy, createRef;
+import 'package:react_testing_library/matchers.dart';
+import 'package:react_testing_library/react_testing_library.dart';
+
+import 'package:test/test.dart';
+import 'package:over_react/components.dart' as components;
+
+import '../component_declaration/builder_integration_tests/new_boilerplate/function_component_test.dart';
+import '../util/prop_conversion_test.dart';
+import '../util/ref_test_cases.dart';
+
+part 'lazy_test.over_react.g.dart';
+
+main() {
+  enableTestMode();
+
+  group('lazy', () {
+    // The fallback and the lazy component render the same thing...
+    uiPropsTest(
+      uiFunction<UiProps>(
+        (props) => (Suspense()..fallback = (Dom.div()..addTestId('testId3'))('id: ${props.id}'))(
+          lazy(
+              () async => uiFunction<UiProps>(
+                    (props2) => (Dom.div()..addTestId('testId3'))('id: ${props.id}'),
+                    UiFactoryConfig(),
+                  ),
+              UiFactoryConfig())()(),
+        ),
+        UiFactoryConfig(),
+      ),
+    );
+
+    group('does not throw an error when', () {
+      test('config is null, IF it only expects GenericUiProps', () {
+        testLazy() => lazy(
+            () async => uiFunction<UiProps>(
+                  (props) => (Dom.div()..addTestId('testId3'))('id: ${props.id}'),
+                  UiFactoryConfig(),
+                ),
+            null);
+        expect(testLazy, returnsNormally);
+        expect(testLazy(), isA<UiFactory<UiProps>>());
```

Comment:
Can we also test the same `initializes the factory variable with a function...` cases as we did below, to make sure this is working properly?